src/main/java/org/mastodon/views/trackscheme/display/PaintHierarchicalGraph.java

```diff
@@ -0,0 +1,156 @@
+ *    and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+package org.mastodon.views.trackscheme.display;
+
+import static org.mastodon.views.trackscheme.ScreenVertex.Transition.DISAPPEAR;
+
+import java.awt.BasicStroke;
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.font.FontRenderContext;
+import java.awt.font.TextLayout;
+import java.awt.geom.Rectangle2D;
+
+import org.mastodon.views.trackscheme.ScreenEntities;
+import org.mastodon.views.trackscheme.ScreenTransform;
+import org.mastodon.views.trackscheme.ScreenVertex;
+import org.mastodon.views.trackscheme.ScreenVertex.Transition;
+import org.mastodon.views.trackscheme.display.style.TrackSchemeStyle;
+
+public class PaintHierarchicalGraph extends PaintBranchGraph
+{
+
+	private final float STROKE_WIDTH = 2.5f;
+
+	private ScreenTransform transform = new ScreenTransform();
+
+	private double averageLetterWidth = 5;
+
+	private double spotRadius;
+
+	private Color labelColor;
+
+	public PaintHierarchicalGraph() {
+		super();
+		this.edgeStroke = new BasicStroke( STROKE_WIDTH );
+	}
+
+	@Override
+	public void paintGraph( Graphics2D g2, ScreenEntities entities, int highlightedVertexId, int highlightedEdgeId, int focusedVertexId, TrackSchemeStyle style )
+	{
+		entities.getScreenTransform( this.transform );
+		super.paintGraph( g2, entities, highlightedVertexId, highlightedEdgeId, focusedVertexId, style );
+	}
+
+	@Override
+	protected void beforeDrawVertices()
+	{
+		g2.setStroke( edgeStroke );
+		calculateAverageLetterWidth();
+		spotRadius = Math.min(transform.getScaleX() * 0.3, transform.getScaleY() * 0.4);
+		labelColor = textColorForBackground( style.getBackgroundColor() );
+	}
+
+	private void calculateAverageLetterWidth()
+	{
+		final FontRenderContext frc = g2.getFontRenderContext();
+		String text = "0123456789abcdefghijklmnopqrstuvwxyz";
+		final TextLayout layout = new TextLayout( text, style.getFont(), frc );
+		averageLetterWidth = layout.getBounds().getWidth() / text.length();
+	}
+
+	@Override
+	protected void drawVertex( ScreenVertex vertex )
+	{
+		if(spotRadius > 3)
+			drawVertexFull( vertex );
+		else
+			drawVertexSimplified( vertex );
+	}
+
```

Comment:
Oh I miss my roundish corners :)
Any chance it could be back, not set by default, and again made configurable in the `TrackSchemeStyle` object?
What do you think?

Reply:
That's a good idea. Let's introduce an option for that.

I prefer the round corners aesthetically. But with the sharp corners I can focus better on the tree structure? Did you observe something similar?

Reply:
..... no. 
I guess it is a matter of taste rather than Science, so it has its place in the settings editor.